### PR TITLE
[BEAM-6679] Fix javadoc @ literals

### PR DESCRIPTION
**Pull request by jklukas**
This change ensures that the core module javadoc:

    ./gradlew ':beam-sdks-java-core:javadoc'

no longer produces the string `@literal` anywhere in the produced html. There may still be some instances in other modules.

Many code examples in the core SDK use `{@literal @}` inside of a `{@code}` block, which does not escape the `@` as expected, but rather produces the text `{@literal @}`.

There's unfortunately not any single solution that's elegant in all cases. The SDK already uses a variety of markup styles for code examples, and I expect that to continue to be the case depending on the complexity of the example.

I've made additional whitespace and typo fixes along with changing how `@literal` is used such that each of these code examples now renders nicely with the current Java 8 build.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](../.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by jklukas** [comment]
R: @echauchot 

This is a follow-up to #7852.

**Comment by jklukas** [comment]
Run Java PreCommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java [hunk 1]
```diff
@@ -277,26 +277,29 @@ private Combine() {
...
    *   }
+   *
    *   public Accum addInput(Accum accum, Integer input) {
    *       accum.sum += input;
    *       accum.count++;
    *       return accum;
    *   }
-   *   public Accum mergeAccumulators(Iterable<Accum> accums) {
+   *
+   *   public Accum{@literal mergeAccumulators(Iterable<Accum> accums)} {
```
Comment (echauchot) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java]: wrong I guess ?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java [hunk 2]
```diff
@@ -277,26 +277,29 @@ private Combine() {
...
    *       merged.count += accum.count;
    *     }
    *     return merged;
    *   }
+   *
    *   public Double extractOutput(Accum accum) {
    *     return ((double) accum.sum) / accum.count;
    *   }
-   * }
+   * }{@literal
```
Comment (echauchot) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java]: wrong ?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java (hunk 1 above)
Comment (jklukas) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java]: In this particular case, we're using `<pre><code>` for the overall block, which doesn't handle `<>`. The `@literal` added here is to accomodate the brackets in `<Accum>`.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java (hunk 2 above)
Comment (jklukas) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java]: Same here, the `@literal` is to handle the angle brackets on the next two lines. We could use `@code` in these two cases instead, but it's redundant with the `<code>` wrapping this entire example.

**Comment by jklukas** [comment]
Run Java PreCommit

**Comment by jklukas** [comment]
Does Jenkins store rendered Javadoc somewhere? I imagine reviewing this would be easier if you had the rendered results to cross-check. My process for creating this PR was to run the gradle javadoc task after changing each class to iterate on getting the whitespace and formatting correct, but I understand if you want to hold off on merging until you have a better basis for validating the changes.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java (hunk 1 above)
Comment (echauchot) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java]: Fair enough as it is rendered well

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java (hunk 2 above)
Comment (echauchot) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Combine.java]: ditto

**Comment by echauchot** [comment]
@jklukas As they are only minor changes I just went through the changes visually and stopped on things that looked odd to me, then I checked out the branch and rendered that javadoc. It renders correctly on these spots. So LGTM, modulo you fix checkstyle

**Comment by jklukas** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by jklukas** [comment]
@echauchot - Apologies for missing the checkstyle errors. Added a new commit to address those and the java precommit is now green. I also rebased on master, and I'm assuming the `Java_Examples_Dataflow` failure is either transient or broken on master.

**Comment by jklukas** [comment]
@echauchot - Apologies for missing the checkstyle errors. Added a new commit to address those and the java precommit is now green. I also rebased on master, and I'm assuming the `Java_Examples_Dataflow` failure is either transient or broken on master.

**Comment by jklukas** [comment]
Run Java_Examples_Dataflow PreCommit

**Comment by jklukas** [comment]
Run Java PreCommit

**Comment by jklukas** [comment]
@echauchot - Squashed to a single commit. Java PreCommit is failing due to an unrelated GrpcDataServiceTest result.

**Pull request merged**